File: producer_with_stubs_per_consumer/src/test/java/com/example/BeerMessagingBase.java
package com.example;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.ImportAutoConfiguration;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.verifier.messaging.boot.AutoConfigureMessageVerifier;
import org.springframework.cloud.stream.binder.test.TestChannelBinderConfiguration;

@SpringBootTest(classes = ProducerApplication.class, webEnvironment = SpringBootTest.WebEnvironment.NONE)
@AutoConfigureMessageVerifier
@ImportAutoConfiguration(TestChannelBinderConfiguration.class)
public abstract class BeerMessagingBase {
	@Autowired
	PersonCheckingService personCheckingService;

	public void clientIsOldEnough() {

		this.personCheckingService.shouldGetBeer(new PersonToCheck(25));

	}

	public void clientIsTooYoung() {

		this.personCheckingService.shouldGetBeer(new PersonToCheck(5));

	}

}

File: producer_kotlin/src/test/kotlin/com/example/BeerMessagingBase.kt
package com.example

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.autoconfigure.ImportAutoConfiguration
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.cloud.contract.verifier.messaging.boot.AutoConfigureMessageVerifier
import org.springframework.cloud.stream.binder.test.TestChannelBinderConfiguration

@SpringBootTest(classes = arrayOf(ProducerApplication::class), webEnvironment = SpringBootTest.WebEnvironment.NONE)
@AutoConfigureMessageVerifier
@ImportAutoConfiguration(TestChannelBinderConfiguration::class)
abstract class BeerMessagingBase {
    @Autowired
    internal var personCheckingService: PersonCheckingService? = null

    fun clientIsOldEnough() {
        this.personCheckingService!!.shouldGetBeer(PersonToCheck(25))
    }

    fun clientIsTooYoung() {
        this.personCheckingService!!.shouldGetBeer(PersonToCheck(5))
    }
}
